pycognaize/model.py

```diff
@@ -379,7 +379,11 @@ def matches(act_tag: Union['ExtractionTag', 'TDTag', 'HTMLTableTag'],
             having the same page number. Returns true if
         intersection is greater than the threshold"""
         if isinstance(act_tag, TDTag) and isinstance(pred_tag, TDTag):
-            return act_tag.html_id == pred_tag.html_id
+            cell_xpath = max(act_tag.xpath, pred_tag.xpath, key=len)
+            field_xpath = min(act_tag.xpath, pred_tag.xpath, key=len)
+            return field_xpath in cell_xpath \
+                and act_tag.col_index == pred_tag.col_index\
+                and act_tag.row_index == pred_tag.row_index
```

Comment:
assign the resulted boolean to some variable and return the variable, so that the return statement is not complicated